README.md

```diff
@@ -6,152 +6,332 @@
+
+# Verify a PDF signature
+./pdfsign verify document.pdf
+
+# Get help for specific commands
+./pdfsign sign -h
+./pdfsign verify -h
+```
+
+## PDF Signing
+
+### Command Line Usage
+
+```bash
+./pdfsign sign [options] <input.pdf> <output.pdf> <certificate.crt> <private_key.key> [chain.crt]
+```
+
+### Signing Options
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `-name` | string | | Name of the signatory |
+| `-location` | string | | Location of the signatory |
+| `-reason` | string | | Reason for signing |
+| `-contact` | string | | Contact information for signatory |
+| `-certType` | string | `CertificationSignature` | Certificate type: `CertificationSignature`, `ApprovalSignature`, `UsageRightsSignature`, `TimeStampSignature` |
+| `-tsa` | string | `https://freetsa.org/tsr` | URL for Time-Stamp Authority |
+
+### Signing Examples
+
+```bash
+# Basic signing
+./pdfsign sign -name "John Doe" input.pdf output.pdf cert.crt key.key
+
+# Signing with additional metadata
+./pdfsign sign -name "John Doe" -location "New York" -reason "Document approval" input.pdf output.pdf cert.crt key.key
+
+# Timestamp-only signature
+./pdfsign sign -certType "TimeStampSignature" input.pdf output.pdf
 ```
-Usage of ./pdfsign:
-  -certType string
-        Type of the certificate (CertificationSignature, ApprovalSignature, UsageRightsSignature, TimeStampSignature) (default "CertificationSignature")
-  -contact string
-        Contact information for signatory
-  -location string
-        Location of the signatory
-  -name string
-        Name of the signatory
-  -reason string
-        Reason for signing
-  -tsa string
-        URL for Time-Stamp Authority (default "https://freetsa.org/tsr")
-
-Example usage:
-        ./pdfsign -name "Jon Doe" sign input.pdf output.pdf certificate.crt private_key.key [chain.crt]
-        ./pdfsign -certType "CertificationSignature" -name "Jon Doe" sign input.pdf output.pdf certificate.crt private_key.key [chain.crt]
-        ./pdfsign -certType "TimeStampSignature" input.pdf output.pdf
-        ./pdfsign verify input.pdf
+
+## PDF Verification
+
+### Command Line Usage
+
+```bash
+./pdfsign verify [options] <input.pdf>
+```
+
+### Verification Options
+
+| Option | Type | Default | Description |
+|--------|------|---------|-------------|
+| `-external` | bool | `false` | Enable external OCSP and CRL checking |
+| `-require-digital-signature` | bool | `true` | Require Digital Signature key usage in certificates |
+| `-require-non-repudiation` | bool | `false` | Require Non-Repudiation key usage in certificates (for highest security) |
+| `-use-signature-time-fallback` | bool | `false` | Use signature time as fallback if no timestamp (untrusted) |
+| `-validate-timestamp-certs` | bool | `true` | Validate timestamp token certificates |
+| `-allow-embedded-roots` | bool | `false` | Allow embedded certificates as trusted roots (use with caution) |
```

Comment:
We should rethink the name of this option as it could be interpreted as that these root are embedded with the application. 

Maybe, `-allow-untrusted-roots`?